docs/email/postfix/postfix-spf-dkim-debian-8.md

```diff
@@ -0,0 +1,376 @@
+
+    mkdir /etc/opendkim
+    mkdir /etc/opendkim/keys
+    chown -R opendkim:opendkim /etc/opendkim
+    chmod go-rw /etc/opendkim/keys
+
+3.  Create the signing table `/etc/opendkim/signing.table`. It needs to have one line per domain you handle mail for, with each line looking like:
+
+{: .file-excerpt}
+/etc/opendkim/signing.table
+:   ~~~ text
+    *@example.com   example
+    ~~~
+
+Replace `example.com` with your domain and `example` with a short name for the domain. The first field is a pattern that matches e-mail addresses, and the second field is a name for the key table entry that should be used to sign mail from that address. For simplicity we're just going to set up one key for all addresses in a domain.
+
+4.  Create the key table `/etc/opendkim/key.table`. It needs to have one line per short domain name in the signing table. Each line will look like:
+
+{: .file-excerpt}
+/etc/opendkim/key.table
+:   ~~~ text
+    example     example.com:YYYYMM:/etc/opendkim/keys/example.private
+    ~~~
+
+Replace `example` with the `example` value you used for the domain in the signing table (make sure to catch the second occurrence at the end where it's followed by `.private`), replace `example.com` with your domain name, and replace the `YYYYMM` with the current 4-digit year and 2-digit month (this is referred to as the selector). The first field connects the signing and key tables. The second field is broken down into 3 sections separated by colons. The first section is the domain name the key is for, the second section is a selector used when looking up key records in DNS, and the third section names the file containing the signing key for the domain.
+
+{: .note}
+>
+> The flow for DKIM lookup starts with the sender's address. The signing table is scanned until an entry whose pattern (first item) matches the address is found, then the second item's value is used to locate the entry in the key table whose key information will be used. For incoming mail the domain and selector are then used to find the public key TXT record in DNS and that public key is used to validate the signature. For outgoing mail the private key is read from the named file and used to generate the signature on the message.
+
+5.  Create the trusted hosts file `/etc/opendkim/trusted.hosts`. It's contents need to be:
+
+{: .file}
+/etc/opendkim/trusted.hosts
+:   ~~~ text
+    127.0.0.1
+    ::1
+    localhost
+    myhostname
+    myhostname.example.com
+    example.com
+    ~~~
+
+When creating the file, change `myhostname` to the name of your server and replace `example.com` with your own domain name. We're identifying the hosts that users will be submitting mail through and that should have their outgoing mail signed, which for basic configurations will just be your own mail server.
+
+6.  Generate keys for each domain. First do a `cd /etc/opendkim/keys`. Then issue the following command:
+
+    opendkim-genkey -b 2048 -r -s YYYYMM
```

Comment:
It always uses example.com in the .txt file, which is just a snippet of a BIND zone file and gets used to set up DNS. The next step snips out just the record itself from between the parentheses and for Linode's DNS Manager you paste it into the form field as the record contents, the comment containing the domain name gets left behind.

The .private file with the private key should be a base64-encoded RSA key file. with the marker lines at the start and end.